ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -154,83 +274,428 @@ class SignalProducerSpec: QuickSpec {
 		}
 
 		describe("startWithSignal") {
-			pending("should invoke the closure before any effects or events") {
+			it("should invoke the closure before any effects or events") {
+				var started = false
+				var value: Int?
+
+				SignalProducer<Int, NoError>(value: 42)
+					|> on(started: {
+						started = true
+					}, next: {
+						value = $0
+					})
+					|> startWithSignal { _ in
+						expect(started).to(beFalsy())
+						expect(value).to(beNil())
+					}
+
+				expect(started).to(beTruthy())
+				expect(value).to(equal(42))
 			}
 
-			pending("should dispose of added disposables if disposed") {
+			it("should dispose of added disposables if disposed") {
+				let addedDisposable = SimpleDisposable()
+				var disposable: Disposable!
+
+				let producer = SignalProducer<Int, NoError>() { _, disposable in
+					disposable.addDisposable(addedDisposable)
+					return
+				}
+
+				producer |> startWithSignal { _, innerDisposable in
+					disposable = innerDisposable
+				}
+
+				expect(addedDisposable.disposed).to(beFalsy())
+
+				disposable.dispose()
+				expect(addedDisposable.disposed).to(beTruthy())
 			}
 
-			pending("should send interrupted if disposed") {
+			it("should send interrupted if disposed") {
+				var interrupted = false
+				var disposable: Disposable!
+
+				SignalProducer<Int, NoError>(value: 42)
+					|> startOn(TestScheduler())
```

Comment:
Only looked briefly but without this `startOn` the signal observer doesn't receive the interrupted event.